server/src/main/java/io/crate/planner/statement/CopyFromPlan.java

```diff
@@ -118,9 +124,15 @@ public void executeOrFail(DependencyCarrier dependencies,
         NodeOperationTree nodeOpTree = NodeOperationTreeGenerator
             .fromPlan(plan, dependencies.localNodeId());
 
-        dependencies.phasesTaskFactory()
-            .create(plannerContext.jobId(), List.of(nodeOpTree))
-            .execute(consumer, plannerContext.transactionContext());
+        JobLauncher jobLauncher = dependencies.phasesTaskFactory()
+            .create(plannerContext.jobId(), List.of(nodeOpTree));
+
+        if (waitForCompletion) {
+            jobLauncher.execute(consumer, plannerContext.transactionContext());
+        } else {
+            jobLauncher.execute(new EmptyRowConsumer(), plannerContext.transactionContext());
+            consumer.accept(InMemoryBatchIterator.of(Row1.ROW_COUNT_UNKNOWN, SENTINEL), null);
```

Comment:
`jobLauncher.execute` might call passed consumer's `accept` in case of exception, see line `consumer.accept(null, throwable)`. @seut, I'm not sure but is it fine in that case, that we are accepting 2 consumers one after one?